lib/src/usage_impl_io.dart

```diff
@@ -87,11 +87,11 @@ class IOPostHandler extends PostHandler {
 
     if (_client == null) {
       _client = mockClient ?? HttpClient();
-      _client.userAgent = _userAgent;
+      _client!.userAgent = _userAgent;
```

Comment:
Seems odd to need the bang right after an assignment to `_client`... Isn't it sure to be non-null here?

Reply:
yup, the fields-don't-promote thing